internal/rbd/rbd_util.go

```diff
@@ -768,16 +812,19 @@ func imageFeaturesToUint64(ctx context.Context, imageFeatures string) uint64 {
 	features := uint64(0)
 
 	for _, f := range strings.Split(imageFeatures, ",") {
```

Comment:
You can use `librbd.FeatureSetFromNames(strings.Split(imageFeatures, ",")` and get the int value directly. Use `Names()` to get a `[]string` with the features that are recognized and see if it misses any of the original requested features.

Also, the strings for features are available like `FeatureNameDeepFlatten`.

Doing it this way, even features that are not known to ceph-csi can get enabled (by advanced users).

Reply:
maybe in followup PR, as I just added one more case  to it.